pallets/liquidity-provider/src/lib.rs

```diff
@@ -63,33 +76,54 @@ decl_module! {
             asset: Asset,
             balance: Balance<T>,
             offer_rates: Vec<OfferRateTy<T>>
-        ) -> dispatch::DispatchResult
+        ) -> DispatchResult
         {
             match asset {
-                Asset::Usdv => return Err(crate::Error::<T>::MustNotBeUsdv.into()),
+                Asset::Usdv => Err(crate::Error::<T>::MustNotBeUsdv.into()),
                 Asset::Collateral(collateral) => {
                     let who = ensure_signed(origin)?;
-                    update_account_rates::<T>(&who, asset, offer_rates);
-                    do_attest::<T>(who.clone(), Asset::Usdv, balance)?;
+                    Self::update_account_rates(&who, asset, offer_rates);
+                    Self::do_attest(who.clone(), Asset::Usdv, balance)?;
                     T::Collateral::deposit(collateral.into(), &who, balance)?;
                     T::Collateral::reserve(collateral.into(), &who, balance)?;
                     Self::deposit_event(RawEvent::Attestation(who, collateral.into()));
                     Ok(())
+                },
+                Asset::Btc | Asset::Cop | Asset::Ves => {
+                    todo!()
                 }
             }
         }
 
+        #[weight = 0]
+        pub fn submit_pair_prices(
+            origin,
+            pair_prices: Vec<PairPrice<Balance<T>>>,
+            _signature: T::Signature,
```

Comment:
> shouldn't it be an ensure_signed transaction

Just the payload is signed for validation purposes. The transaction itself isn't signed because it would enforce a fee for every node as described in `Unsigned transactions`.